modules/data.atmosphere/R/download.AmerifluxLBL.R

```diff
@@ -14,50 +17,79 @@
 ##' result <- download.AmerifluxLBL("US-Akn","~/","2011-01-01","2011-12-31",overwrite=TRUE)
 ##' }
 ##' 
-##' @author Ankur Desai, based on download.Ameriflux.R by Josh Mantooth, Rob Kooper, Shawn Serbin
-download.AmerifluxLBL <- function(sitename, outfolder, start_date, end_date, 
-                                  overwrite = FALSE, verbose = FALSE, username = "pecan", method, ...) {
-  # get start/end year code works on whole years only
-  
-  
-  site <- sub(".* \\((.*)\\)", "\\1", sitename)
+##' @author Ankur Desai, Henri Kajasilta based on download.Ameriflux.R by Josh Mantooth, Rob Kooper, Shawn Serbin
+
+
+download.AmerifluxLBL <- function(sitename, outfolder, start_date, end_date,
+                                  overwrite = FALSE, verbose = FALSE, username = "pecan", method,
+                                  useremail = "@", data_product = "BASE-BADM", data_policy = "CCBY4.0", ...) {
   
+  # Initial set-ups for amerifluxr packages
+  # get start/end year code works on whole years only
   start_date <- as.POSIXlt(start_date, tz = "UTC")
   end_date <- as.POSIXlt(end_date, tz = "UTC")
   
   start_year <- lubridate::year(start_date)
   end_year <- lubridate::year(end_date)
   
+  
+  site <- sub(".* \\((.*)\\)", "\\1", sitename)
+  
+  
+  
+  
+  
   # make sure output folder exists
   if (!file.exists(outfolder)) {
     dir.create(outfolder, showWarnings = FALSE, recursive = TRUE)
   }
   
-  # need to query to get full file name #this is Ameriflux version
-  url <- "http://wile.lbl.gov:8080/AmeriFlux/DataDownload.svc/datafileURLs"
-  json_query <- paste0("{\"username\":\"", username, "\",\"siteList\":[\"", site, "\"],\"intendedUse\":\"Research - Land model/Earth system model\",\"description\":\"PEcAn download\"}")
-  result <- httr::POST(url, body = json_query, encode = "json", httr::add_headers(`Content-Type` = "application/json"))
-  link <- httr::content(result)
   
-  ftplink <- NULL
-  if (length(link$dataURLsList) > 0) {
-    ftplink <- link$dataURLsList[[1]]$URL
+
+  repeat {
+    tout <- options("timeout")
+    zip_file <- try(amerifluxr::amf_download_base(user_id = username,
+                                                  user_email = useremail,
+                                                  site_id = site,
+                                                  data_product = data_product,
+                                                  data_policy = data_policy,
+                                                  agree_policy = TRUE,
+                                                  intended_use = "model",
+                                                  intended_use_text = "PEcAn download",
+                                                  verbose = verbose,
+                                                  out_dir = outfolder)
+    )
+    if (!inherits(zip_file, "try-error")){
+      break
+    }else if(tout$timeout > 250 ){
+      PEcAn.logger::logger.info("Download takes too long, check your connection.")
```

Comment:
```suggestion
      PEcAn.logger::logger.severe("Download takes too long, check your connection.")
```
Maybe this should stop here already